src/libstore/build/worker.cc

```diff
@@ -380,7 +380,10 @@ void Worker::waitForInput()
         std::set<int> fds2(j->fds);
         std::vector<unsigned char> buffer(4096);
         for (auto & k : fds2) {
-            if (pollStatus.at(fdToPollStatus.at(k)).revents) {
+            auto fdps1 = fdToPollStatus.find(k);
+            if (fdToPollStatus.end() == fdps1)
+                throw Error("fdToPollStatus lookup failed for fd %1%", k);
+            if (pollStatus.at(fdps1->second).revents) {
```

Comment:
this one is `vector::at`, it's distinct from the `map::at` (`map::at` is more difficult to debug, because the key is more complex than an index). But ok, this one should imo an assert instead.